pkg/asset/imagebased/image/ignition.go

```diff
@@ -0,0 +1,183 @@
+package image
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/coreos/ignition/v2/config/merge"
+	"github.com/coreos/ignition/v2/config/v3_2"
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	ibiConfigurationPath     = "/var/tmp/ibi-configuration.json"
+	installationScriptPath   = "/usr/local/bin/install-rhcos-and-restore-seed.sh"
+	installationServiceName  = "install-rhcos-and-restore-seed.service"
+	networkConfigServiceName = "network-config.service"
+	pullSecretPath           = "/var/tmp/pull-secret.json" //nolint:gosec // not a secret, just a filename
+
+	trustedBundlePath        = "/etc/pki/ca-trust/source/anchors/additional-trust-bundle.pem"
+	nmstateConfigPath        = "/var/tmp/network-config.yaml"
+	registriesConfPath       = "/etc/containers/registries.conf"
+	postDeploymentScriptPath = "/var/tmp/post.sh"
+)
+
+// Ignition is an asset that generates the image-based installer ignition file.
+type Ignition struct {
+	Config *igntypes.Config
+}
+
+// Name returns the human-friendly name of the asset.
+func (i *Ignition) Name() string {
+	return "Image-based Installer Ignition"
+}
+
+// Dependencies returns the assets on which the Ignition asset depends.
+func (i *Ignition) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&ImageBasedInstallationConfig{},
+		&InstallationService{},
+		&RegistriesConf{},
+		&PostDeployment{},
+	}
+}
+
+type ibiConfigurationFile struct {
+	ExtraPartitionLabel  string `json:"extraPartitionLabel,omitempty"`
+	ExtraPartitionNumber uint   `json:"extraPartitionNumber,omitempty"`
+	ExtraPartitionStart  string `json:"extraPartitionStart,omitempty"`
+	InstallationDisk     string `json:"installationDisk"`
+	PrecacheBestEffort   bool   `json:"precacheBestEffort,omitempty"`
+	PrecacheDisabled     bool   `json:"precacheDisabled,omitempty"`
+	SeedImage            string `json:"seedImage"`
+	SeedVersion          string `json:"seedVersion"`
+	Shutdown             bool   `json:"shutdown,omitempty"`
+	SkipDiskCleanup      bool   `json:"skipDiskCleanup,omitempty"`
+	UseContainersFolder  bool   `json:"useContainersFolder,omitempty"`
```

Comment:
Lets remove this one as this is more like for future, currently we don't support folder